suite-native/module-device-onboarding/src/screens/DeviceAuthenticityScreen.tsx

```diff
@@ -0,0 +1,51 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+import { selectSelectedDeviceAuthenticity } from '@suite-common/wallet-core';
+import { ContinueOnTrezorScreenContent, useDeviceAuthenticityCheck } from '@suite-native/device';
+import {
+    DeviceOnboardingStackParamList,
+    DeviceOnboardingStackRoutes,
+    StackProps,
+} from '@suite-native/navigation';
+
+import { DeviceOnboardingScreenWithExitButton } from '../components/DeviceOnboardingScreenWithExitButton';
+
+export const DeviceAuthenticityScreen = ({
+    navigation,
+}: StackProps<DeviceOnboardingStackParamList, DeviceOnboardingStackRoutes.DeviceAuthenticity>) => {
+    const [isCheckActive, setIsCheckActive] = useState(false);
+    const { checkDeviceAuthenticity } = useDeviceAuthenticityCheck();
+    const selectedDeviceAuthenticity = useSelector(selectSelectedDeviceAuthenticity);
+
+    // Soft error meaning we are not sure if the device is compromised.
+    // We need to retry, and the user cannot proceed past this screen,
+    // but we do not report this to the user as a compromised device.
+    // This includes actions such as canceling on the device or disconnecting the device.
+    const hasSoftError =
```

Comment:
Nit: I'd add explicit declaration. Imagine someone in future makes a mistake, removes the `!!` → the conjunction shortcircuits to an object → triggering `useEffect` repeatedly → unpredictable cycles.

:thought_balloon:  Still, I don't really like how we're relying on `checkDeviceAuthenticity` to clear the error (it must, so `hasSoftError` resets to `false`, so if the check fails _again_, it will then be set _again_ to `true`. If it stayed `true` the whole time, there'd be no third attempt).
But with this logic I don't see a low-hanging fruit to refactor it. At least, if we can make sure it's `boolean`, and therefore referentially stable, it will be predictable. 
```suggestion
    const hasSoftError: boolean =
```

Reply:
8d3e060ee416acb4c00f2be3fb026ccbc8c5a8a0